lib/node/runner/checker_ballot_transaction.go

```diff
@@ -25,6 +28,41 @@ type BallotTransactionChecker struct {
 	ValidTransactions     []string
 	validTransactionsMap  map[string]bool
 	CheckTransactionsOnly bool
+	transactionsCache     map[string]transaction.Transaction
+}
+
+func (b *BallotTransactionChecker) getTransaction(hash string) (tx transaction.Transaction, found bool, err error) {
+	b.RLock()
+	tx, found = b.transactionsCache[hash]
+	b.RUnlock()
+
+	if found {
+		return
+	}
+
+	b.Lock()
+	defer b.Unlock()
+
+	tx, found = b.NodeRunner.TransactionPool.Get(hash)
```

Comment:
@spikeekips I have a question about this closed PR.
`b.transactionsCache` is superset of `runner.TransactionPool` and `block.TranscationPool`, right?
Then some tx can be in both. Isn't it a waste of memory?

Reply:
@Charleslee522 Yes, it will be, but `transactionCache` will be removed after checking. It will help `TransactionPool` to be clean.

Reply:
@spikeekips Thanks. Clear!